basic-javascript/project-2.js

```diff
@@ -68,27 +76,33 @@ const isPrime = num => {
 };
 
 const returnFirst = arr => {
-  return arr[0];
+  return arr === undefined ? null : arr[0];
 };
 
 const returnLast = arr => {
-  return arr[arr.length - 1];
+  return arr === undefined ? null : arr[arr.length - 1];
 };
 
 const getArrayLength = arr => {
-  return arr.length;
+  return arr === undefined ? null : arr.length;
 };
 
 const incrementByOne = arr => {
-  for (let i = 0; i < arr.length; i++) {
-    arr[i]++;
+  if (arr !== undefined) {
+    for (let i = 0; i < arr.length; i++) {
+      arr[i]++;
+    }
+    return arr;
   }
-  return arr;
 };
 
 const addItemToArray = (arr, item) => {
-  arr.push(item);
-  return arr;
+  if (arr === undefined) {
+    return null;
```

Comment:
You should check if the type of arr is actually an array. If you make `arr` a string, it is still defined and it will try to `.push` the item onto it.